### PR TITLE
Refactor Conditions to Fields for ASA rule definition

### PR DESCRIPTION
**Pull request by zhang-hua**
# Type of change? 

- [ ] Bug fix
- [ ] New feature
- [x] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 

Refactor Conditions to Fields which containers a string array of measurement names.

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by zhang-hua** [comment]
ARM template deployment tests passed.

File: solutions/remotemonitoring/armtemplates/basic-static-map.json [hunk 1]
```diff
@@ -261,7 +261,7 @@
...
...ription as [rule.description],\n        AI.__severity as [rule.severity],\n        AI.__ruleid as [rule.id],\n        AI.__deviceId as [device.id],\n        AI.__aggregates,\n        AI.__receivedTime as [device.msg.received]\n    FROM\n        ApplyInstantRuleFilters AI PARTITION BY __partitionid\n)\n\n-- Output alarm events\nSELECT\n    CA.*\nINTO\n    Alarms\nFROM\n    CombineAlarms CA PARTITION BY __partitionid\n\n-- Output origin telemetry messages\nSELECT\n        CONCAT(T.IoTHub.ConnectionDeviceId, ';', CAST(DATEDIFF(millisecond, '1970-01-01T00:00:00Z', T.EventEnqueuedUtcTime) AS nvarchar(max))) as id,\n        1 as [doc.schemaVersion],\n        'd2cmessage' as [doc.schema],\n        T.IoTHub.ConnectionDeviceId as [device.id],\n        'device-sensors;v1' as [device.msg.schema],\n        'StreamingJobs' as [data.schema],\n        DATEDIFF(millisecond, '1970-01-01T00:00:00Z', System.Timestamp) as [device.msg.created],\n        DATEDIFF(millisecond, '1970-01-01T00:00:00Z', T.EventEnqueuedUtcTime) as [device.msg.received],\n        udf.removeUnusedProperties(T) as Data\nINTO\n    Messages\nFROM\n    DeviceTelemetry T PARTITION BY PartitionId TIMESTAMP BY T.EventEnqueuedUtcTime"
```
Comment (ppathan) [inline solutions/remotemonitoring/armtemplates/basic-static-map.json]: Are we keeping comments here? `-- Aggregate union results for time windows over 1, 5 and 10 minutes`

Comment (zhang-hua) [reply solutions/remotemonitoring/armtemplates/basic-static-map.json]: yes.

**Comment by zhang-hua** [comment]
@ppathan ,  all comments have been removed from query.

**Pull request merged**